### ISSUE TITLE
[WARN] PersistentConnection: pc_0 - Provided authentication credentials are invalid

### ISSUE THREAD

**Issue by winstonma** [firebase/quickstart-java#4]
When I run the function `startWeeklyTopPostEmailer()` it has the following problem.

`[WARN] PersistentConnection: pc_0 - Provided authentication credentials are invalid. This usually indicates your FirebaseApp instance was not initialized correctly. Make sure your google-services.json file has the correct firebase_url and api_key. You can re-download google-services.json from https://console.firebase.google.com/.
`

I created the firebase app and created service account key (JSON).

Please let me know if I need to provide more info. Thanks

**Comment by samtstern** [comment]
The important lines in the quickstart are:
```
            FirebaseOptions options = new FirebaseOptions.Builder()
                    .setServiceAccount(new FileInputStream("service-account.json"))
                    .setDatabaseUrl(DATABASE_URL)
                    .build();
            FirebaseApp.initializeApp(options);
```

This requires that you have properly placed the `service-account.json` file and have updated the `DATABASE_URL` variable.  Have you done both of those? Are there other errors in your logs?

**Comment by winstonma** [comment]
I modified DATABASE_URL in Database.java and placed the service-account.json in database folder (where the gradlew is placed).

Thanks

**Comment by winstonma** [comment]
Just found it that there is some problem in key creation.

When a key is being created, I need to add `Editor` role

**Comment by samtstern** [comment]
@winstonma did adding 'Editor' solve your issue?

**Comment by winstonma** [comment]
Yes I followed this [Stackoverflow Issue](http://stackoverflow.com/questions/37571543/authentication-with-firebase-3-0-in-node-js) and created a new service account.

Originally I created a service account without any role attached to it I got the problem above.

**Comment by samtstern** [comment]
Fixed by #8
